src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -514,6 +516,21 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
       requestMemory = conf.get(ARMADA_EXECUTOR_REQUEST_MEMORY)
     )
 
+    val driverIngress = if (conf.get(ARMADA_SPARK_DRIVER_INGRESS_ENABLED)) {
+      Some(
+        IngressConfig(
+          annotations = conf
+            .get(ARMADA_SPARK_DRIVER_INGRESS_ANNOTATIONS)
+            .map(commaSeparatedAnnotationsToMap)
+            .getOrElse(Map.empty),
+          tls = conf.get(ARMADA_SPARK_DRIVER_INGRESS_TLS_ENABLED),
+          certName = conf.get(ARMADA_SPARK_DRIVER_INGRESS_CERT_NAME)
```

Comment:
Have you taken a look at how the internal GR web ui feature step sets up the cert from a secret?  We'll probably have to allow something similar.